proto/v1/rollout_service.proto

```diff
@@ -101,9 +117,406 @@ message Plan {
+  string target = 6;
+
+  oneof payload {
+    DatabaseCreate database_case = 7;
+    DatabaseSchemaBaseline database_schema_baseline = 8;
+    DatabaseSchemaUpdate database_schema_update = 9;
+    DatabaseDataUpdate database_data_update = 10;
+    DatabaseRestorePitrRestore database_restore_pitr_restore = 11;
+  }
+
+  message DatabaseCreate {
+    // The project owning the database.
+    // Format: projects/{project}
+    string project = 1;
+    // database name
+    string database = 2;
+    // table name
+    string table = 3;
+    // Format: sheets/{sheet}
+    string sheet = 4;
+    string character_set = 5;
+    string collection = 6;
+    map<string, string> labels = 9;
+  }
+
+  message DatabaseSchemaBaseline {
+    string schema_version = 1;
+  }
+
+  message DatabaseSchemaUpdate {
+    // Format: sheets/{sheet}
+    string sheet = 1;
+    string schema_version = 2;
+  }
+
+  message DatabaseDataUpdate {
+    // Format: sheets/{sheet}
+    string sheet = 1;
+    string schema_version = 2;
+
+    // MySQL rollback SQL related.
+
+    // Build the rollback SQL if rollback_enabled.
+    bool rollback_enabled = 3;
+    enum RollbackSqlStatus {
+      ROLLBACK_SQL_STATUS_UNSPECIFIED = 0;
+      PENDING = 1;
+      DONE = 2;
+      FAILED = 3;
+    }
+    // The status of the rollback SQL generation.
+    RollbackSqlStatus rollback_sql_status = 4;
+    // transaction_id is the ID of the transaction executing the migration.
+    // It is only use for Oracle to find Rollback SQL statement now.
+    string transaction_id = 5;
+    // thread_id is the ID of the connection executing the migration.
+    // We use it to filter the binlog events of the migration transaction.
+    string thread_id = 6;
+    // change_history is the resource name of the change history.
+    // We use it to get the schema when the transaction ran.
+    // Format: instances/{instance}/databases/{database}/changeHistories/{changeHistory}
+    string change_history = 7;
+    string binlog_file_start = 8;
+    string binlog_file_end = 9;
+    int64 binlog_position_start = 10;
+    int64 binlog_position_end = 11;
+    string rollback_error = 12;
+    // rollback_sheet is the resource name of
+    // the sheet that stores the generated rollback SQL statement.
+    // Format: sheets/{sheet}
+    string rollback_sheet = 13;
+    // rollback_from_review is the resource name of the review that
+    // the rollback SQL statement is generated from.
+    // Format: projects/{project}/reviews/{review}
+    string rollback_from_review = 14;
+    // rollback_from_task is the resource name of the task that
+    // the rollback SQL statement is generated from.
+    // Format: projects/{project}/rollouts/{rollout}/stages/{stage}/tasks/{task}
+    string rollback_from_task = 15;
+  }
+
+  message DatabaseBackup {
+    // The resource name of the backup.
+    // FIXME: format TBD.
+    string backup = 1;
+  }
+
+  message DatabaseRestorePitrRestore {
```

Comment:
DatabaseRestore since it covers both full restore and pitr?

Reply:
The name corresponds to the task type, which is `DATABASE_RESTORE_PITR_RESTORE`

Reply:
We use these two tasks,`DATABASE_RESTORE_PITR_RESTORE` and `DATABASE_RESTORE_PITR_CUTOVER`, to restore databases.